code/modules/reagents/chemistry/recipes.dm

```diff
@@ -300,8 +300,30 @@
 		var/datum/effect_system/reagents_explosion/e = new()
 		e.set_up(power , T, 0, 0)
 		e.start(holder.my_atom)
-	holder.clear_reagents()
-
+	if (ismob(holder.my_atom) && clear_mob_reagents)
+		/// Only clear reagents if they use a special explosive reaction to do it; it shouldn't apply
+		/// to any explosion inside a person
+		holder.clear_reagents()
+		if(iscarbon(holder.my_atom))
+			var/mob/living/carbon/victim = holder.my_atom
+			var/vomit_flags = MOB_VOMIT_MESSAGE | MOB_VOMIT_FORCE
+			/// The vomiting here is for effect, not meant to help with purging
+			victim.vomit(vomit_flags, distance = 5)
```

Comment:
5 is an enormous distance, maybe scale with amount purged?

Reply:
It's meant to deliberately be a ridiculous spectacle of vomit.